internal/jujucommands/jujucommands.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2025 Canonical.
+
+// Package jujucommands provides functions run juju cmds from a JIMM instance.
+// Each command function is run with its own isolated in-mem store.
+
+package jujucommands
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/juju/cmd/v3"
+	"github.com/juju/juju/cmd/juju/commands"
+	"github.com/juju/juju/jujuclient"
+	"github.com/mitchellh/go-linereader"
+)
+
+var (
+	whitelist = []string{"help", "bootstrap"}
+)
+
+type outputLine struct {
+	Line string
+	Err  error
+}
+
+// runCmdWithOutputRetriever runs the command given by cmdAndArgs using the provided jujuclient.ClientStore.
```

Comment:
This command would be more helpful on a function/method that is exported.
Should this function be exported? Or do we expect to have something else wrapping this?

Reply:
No no not exported. The comment was more to help understand how to use it. Commands in this package will use this.